Mathlib/InformationTheory/BinaryEntropy.lean

```diff
@@ -0,0 +1,345 @@
+/-
+Copyright (c) 2023 Adomas Baliuka. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adomas Baliuka
+-/
+import Mathlib.Analysis.SpecialFunctions.Log.Base
+import Mathlib.Analysis.SpecialFunctions.Log.NegMulLog
+import Mathlib.Analysis.Convex.SpecificFunctions.Basic
+
+/-!
+# Properties of Shannon binary entropy
+
+The [binary entropy function](https://en.wikipedia.org/wiki/Binary_entropy_function)
+`h₂ p := - p * log₂ p - (1-p) * log₂ p`
+is the Shannon entropy of a Bernoulli random variable with success probability `p`.
+
+This file assumes that entropy is measured in bits, hence logarithms base 2.
+Most lemmas are also valid using a different-base logarithm.
+
+## Tags
+
+entropy, Shannon, binary
+-/
+
+namespace Entropy
+
+open Real
+
+/-- Base-2 logarithm-/
+noncomputable abbrev log₂ (p : ℝ) : ℝ := Real.logb 2 p
+
+lemma differentiableAt_log₂ {x : ℝ} (hx : x ≠ 0): DifferentiableAt ℝ log₂ x := by
+    unfold log₂ logb
+    apply DifferentiableAt.div_const (DifferentiableAt.log differentiableAt_id' hx)
+
+/-- Shannon Binary entropy function (measured in bits).
+It's the Shannon entropy of a Bernoulli random variable with success probability `p`.
+Usual domain of definition is p ∈ [0,1], i.e., input is a probability.
+`h₂ p := - p * log₂ p - (1-p) * log₂ p` -/
+noncomputable def h₂ (p : ℝ) : ℝ := -p * log₂ p - (1 - p) * log₂ (1 - p)
```

Comment:
Should be a longer name, e.g. `shannonEntropy`, or perhaps `Entropy.shannon` is okay. Although maybe that should be reserved for the entropy of something more general.

In any case, `h\_2` is not okay as a definition name, I think.